### PR TITLE
Added modified knapsack algorithm.

### PR DESCRIPTION
**Pull request by shivhek25**
Solves issue #134

### CODE REVIEW

File: Algorithms/Dynamic-Programming/modified_knapsack.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+			}
+			// if weight of current item exceeds knapsack weight in consideration, we simply take
+			// the best case for 1 the items that have been checked already for the same knapsack capacity
+			else{
+            			K[i][w].maxval = K[i-1][w].maxval;
+            			K[i][w].maxitems = K[i-1][w].maxitems;
+			}
+		}
+	}
+	cout<< "Maximum weight possible : " << K[n][W].maxval << endl;
```
Comment (ramitsawhney27) [inline Algorithms/Dynamic-Programming/modified_knapsack.cpp]: Do you mean maximum profit/value possible?

Comment (shivhek25) [reply Algorithms/Dynamic-Programming/modified_knapsack.cpp]: Yes, mentioned weight by mistake. I'll make that change. Is there any other change that I need to make ?

**Comment by shivhek25** [comment]
I mentioned maximum weight possible by mistake. It's actually the maximum profit / value possible.

**Comment by ramitsawhney27** [comment]
Could you run your code for this input?

3
1 1 1 1 1 2
3 2

**Comment by shivhek25** [comment]
@ramitsawhney27 I realised where my code went wrong. I need to add a check for maximum value among all entries in the dp table such that i==n or j==W, while printing. I will make the change and commit soon.

**Comment by ramitsawhney27** [comment]
It’s more natural to code using top-down memoization rather than bottom-up. Ideally you should be considering O(nkw) cases rather than using some sort of heuristic with O(nw) heuristics. It’s a much smaller and natural solution that way. Even if you add a check for the maximum value, can you prove you’ll get the optimal answer? If yes, add one.

**Comment by shivhek25** [comment]
@ramitsawhney27 I had worked out the cases when the limit was being exceeded or not, and had coded it accordingly. I have added the explanation.

**Comment by shivhek25** [comment]
@ramitsawhney27 Could you kindly review the changes ?

**Pull request merged**